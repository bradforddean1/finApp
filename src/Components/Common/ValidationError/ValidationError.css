.ValidationError {
	margin-bottom: 12px;
	height: 12px;
	font-family: Sofia Pro;
	font-style: normal;
	font-weight: normal;
	font-size: 12px;
	line-height: 12px;
	color: #ff0000;
}
