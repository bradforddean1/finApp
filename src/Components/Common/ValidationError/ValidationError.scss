.ValidationError {
	margin-bottom: 12px;
	height: 12px;
	font-family: "Poppins";
	font-style: normal;
	font-weight: normal;
	font-size: 12px;
	line-height: 12px;
	color: #ff0000;
}
