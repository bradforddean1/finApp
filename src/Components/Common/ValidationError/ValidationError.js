import React from "react";
import "./ValidationError.css";
import PropTypes from "prop-types";

/**
 * Validation Error for Form Elements
 * @component
 */
function ValidationError(props) {
	const { children } = props;
	return <div className="ValidationError">{children}</div>;
}

ValidationError.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.element,
		PropTypes.arrayOf(PropTypes.element),
	]),
};

export default ValidationError;
