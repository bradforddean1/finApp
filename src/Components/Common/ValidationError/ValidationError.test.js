import React from "react";
import ReactDOM from "react-dom";
import ValidationError from "./ValidationError";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("ValidationError Component", () => {
	it("Renders", () => {
		const div = document.createElement("div");
		ReactDOM.render(<ValidationError />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("Renders the Default State", () => {
		const wrapper = shallow(<ValidationError />);
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Renders given props", () => {
		const wrapper = shallow(<ValidationError>Error</ValidationError>);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
