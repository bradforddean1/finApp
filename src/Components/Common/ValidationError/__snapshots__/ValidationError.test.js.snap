// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidationError Component Renders given props 1`] = `
<div
  className="ValidationError"
>
  Error
</div>
`;

exports[`ValidationError Component Renders the Default State 1`] = `
<div
  className="ValidationError"
/>
`;
